app/validators/rules/rule_validator.py

```diff
@@ -0,0 +1,207 @@
+from app.validators.validator import Validator
+
+
+class Operator:
+    NOT = "not"
+    AND = "and"
+    OR = "or"
+    EQUAL = "=="
+    NOT_EQUAL = "!="
+    GREATER_THAN = ">"
+    LESS_THAN = "<"
+    GREATER_THAN_OR_EQUAL = ">="
+    LESS_THAN_OR_EQUAL = "<="
+    IN = "in"
+    ALL_IN = "all-in"
+    ANY_IN = "any-in"
+    COUNT = "count"
+    DATE = "date"
+    FORMAT_DATE = "format-date"
+    DATE_RANGE = "date-range"
+    MAP = "map"
+
+
+LOGIC_OPERATORS = [Operator.NOT, Operator.AND, Operator.OR]
+
+COMPARISON_OPERATORS = [
+    Operator.EQUAL,
+    Operator.NOT_EQUAL,
+    Operator.GREATER_THAN,
+    Operator.LESS_THAN,
+    Operator.GREATER_THAN_OR_EQUAL,
+    Operator.LESS_THAN_OR_EQUAL,
+]
+
+ARRAY_OPERATORS = [Operator.IN, Operator.ALL_IN, Operator.ANY_IN]
+
+VALUE_OPERATORS = [
+    Operator.DATE,
+    Operator.COUNT,
+    Operator.FORMAT_DATE,
+    Operator.DATE_RANGE,
+    Operator.MAP,
+]
+
+ALL_OPERATORS = (
+    LOGIC_OPERATORS + COMPARISON_OPERATORS + ARRAY_OPERATORS + VALUE_OPERATORS
+)
+
+
+SELF = "self"
```

Comment:
Self is a bit awkward when used with python instances. I can just about follow it, but my brain did get muddled if it was the class instance or the answer etc

Reply:
I could rename the const itself to like `SELF_REFERENCE_KEY` or something like that if it is confusing. Class instance would never be upper case so in that sense I think it is okay but not fussed, so happy to rename for clarity if required.